cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceLifecycleContext.java

```diff
@@ -126,6 +127,16 @@ public String getNamespace() {
   }
 
   @Override
+  public void addLocalFile(URI uri) {
+    throw new UnsupportedOperationException("Local cache files are not supported in Mapper/Reducer tasks.");
```

Comment:
I see what you mean. Yes, this should just delegate, and ``BasicMapReduceTaskContext`` should handle it. Will fix.